<?xml version="1.0" ?>
<sdf version="1.10">
    <world name="empty_world">

        <physics name="1ms" type="ignored">  <!--default from gazebo sim docs-->
            <max_step_size>0.001</max_step_size>
            <real_time_factor>1.0</real_time_factor>
        </physics>
        <plugin
            filename="gz-sim-physics-system"
            name="gz::sim::systems::Physics">
        </plugin>
        <plugin
            filename="gz-sim-user-commands-system"
            name="gz::sim::systems::UserCommands">
        </plugin>  <!--default from gazebo sim docs-->
        <plugin
            filename="gz-sim-scene-broadcaster-system"
            name="gz::sim::systems::SceneBroadcaster">
        </plugin>  <!--default from gazebo sim docs-->

        <plugin name="gz::sim::systems::ApplyLinkWrench" filename="gz-sim-apply-link-wrench-system" />
        <plugin name="gz::sim::systems::PosePublisher" filename="gz-sim-pose-publisher-system">
            <publish_nested_model_pose>true</publish_nested_model_pose>
        </plugin>
        <plugin name="gz::sim::systems::Sensors" filename="gz-sim-sensors-system"><render_engine>ogre2</render_engine></plugin>

        <light type="directional" name="sun">  <!--default from gazebo sim docs-->
            <cast_shadows>true</cast_shadows>
            <pose>0 0 10 0 0 0</pose>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.2 0.2 0.2 1</specular>
            <attenuation>
                <range>1000</range>
                <constant>0.9</constant>
                <linear>0.01</linear>
                <quadratic>0.001</quadratic>
            </attenuation>
            <direction>-0.5 0.1 -0.9</direction>
        </light>

        <model name="ground_plane">  <!--default from gazebo sim docs-->
            <static>true</static>
            <link name="link">
                <collision name="collision">
                <geometry>
                    <plane>
                    <normal>0 0 1</normal>
                    </plane>
                </geometry>
                </collision>
                <visual name="visual">
                <geometry>
                    <plane>
                    <normal>0 0 1</normal>
                    <size>100 100</size>
                    </plane>
                </geometry>
                <material>
                    <ambient>0.8 0.8 0.8 1</ambient>
                    <diffuse>0.8 0.8 0.8 1</diffuse>
                    <specular>0.8 0.8 0.8 1</specular>
                </material>
                </visual>
            </link>
        </model>

        <gui fullscreen="0">  <!--default from gazebo sim docs-->
            <plugin filename="MinimalScene" name="3D View">
              <gz-gui>
                <title>3D View</title>
                <property type="bool" key="showTitleBar">false</property>
                <property type="string" key="state">docked</property>
              </gz-gui>

              <engine>ogre2</engine>
              <scene>scene</scene>
              <ambient_light>0.4 0.4 0.4</ambient_light>
              <background_color>0.8 0.8 0.8</background_color>
              <camera_pose>50 50 50 0 0.5 3.95</camera_pose>  <!--camera forward axis is x, rotation in radians-->
              <camera_clip>
                <near>0.25</near>
                <far>25000</far>
              </camera_clip>
            </plugin>
            <plugin filename="GzSceneManager" name="Scene Manager">
              <gz-gui>
                <property key="resizable" type="bool">false</property>
                <property key="width" type="double">5</property>
                <property key="height" type="double">5</property>
                <property key="state" type="string">floating</property>
                <property key="showTitleBar" type="bool">false</property>
              </gz-gui>
            </plugin>
        </gui>

    </world>
</sdf>